### PR TITLE
Code generation

### LINKED ISSUE #55: Add CodeGenerator for structural ontologies
Atm., the F# code for structural ontologies are written by hand from the respective OBO/YAML files. We need to automatize this for easier updating of the structural ontology libraries.

### PR DESCRIPTION
**Pull request by omaus**
This PR
- adds functions for code generation (from OBO files to F# source files)
  - unit tests accordingly
- closes #55 

Left to do:
- (GitHub action for automatized execution?)

### CODE REVIEW

File: codeGenTestScript.fsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+let toCodeString (term : OboTerm) = 
+    $"    let {toUnderscoredName term} = CvTerm.create(\"{term.Id}\", \"{term.Name}\", \"{toTermSourceRef term}\"){System.Environment.NewLine}{System.Environment.NewLine}"
+
+toCodeString onto.Terms.Head
+
+let toSourceCode (onto : OboOntology) =
+    let concattedSingleValues = String.init onto.Terms.Length (fun i -> $"{toCodeString onto.Terms[i]}")
+    $"{baseString}{concattedSingleValues}"
+
+toSourceCode onto
```
Comment (kMutagene) [inline codeGenTestScript.fsx:42]: Please stop committing test scripts to repos. Pls refactor anything relevant to actual unit tests or do not commit it at all.

File: playground.fsx [hunk 2]
```diff
@@ -124,7 +133,7 @@ let expectedTermValuesSimple =
         [""]
         [""]
         [""]
-        [""; "Maus"; "Keider"; "m�ller"; "oih"]
+        [""; "Maus"; "Keider"; "müller"; "oih"]
```
Comment (kMutagene) [inline playground.fsx:136]: you seriously need to fix your encoding problems 😆

**Review by kMutagene — CHANGES_REQUESTED** [review]
Looks good so far, with minor changes.

Also, please incorporate code generation into the build process, so that the `.fs` files for the ontologies get generated the same way we currently generate the `.obo` files from `yml` 
see https://github.com/nfdi4plants/ARCTokenization/blob/e5280af758f71d54c0414f8c2af492d8bc7cedf4/build/BasicTasks.fs#L37

This most likely means that the code generation must move into a separate project, as we must execute it before building the other projects that should contain the generated `.fs` files. ideally, make it a console application so we can just build and run it via a fake process

File: build/BasicTasks.fs [hunk 3]
```diff
@@ -6,6 +6,8 @@ open Fake.DotNet
 open Fake.IO.Globbing.Operators
 open System.IO
 open Fake.IO
+open ARCTokenization.StructuralOntology
```
Comment (kMutagene) [inline build/BasicTasks.fs:9]: While it seems to work, i really do not like this circular dependency, where the build project is dependent on one of the projects it is supposed to build.

**Review by kMutagene — CHANGES_REQUESTED** [review]
My initial idea for this has been adding a separate runnable project (basically the same type as the build project) and invoking that from within the build project's pipelines.

However, looking at this PR, i think it would be best to set up OBO.NET.CodeGeneration as a completely separate library. This way, we can cleanly add a dependency to that package in the build project, keep the rest as-is, and also make code generation from obo files usable in other projects. Additionally, we do not bloat this repo further with another project.

I think the only thing to adapt code-wise would be parameterizing the namespace of the generated files.

What do you think @omaus?

**Comment by omaus** [comment]
> My initial idea for this has been adding a separate runnable project (basically the same type as the build project) and invoking that from within the build project's pipelines.
> 
> However, looking at this PR, i think it would be best to set up OBO.NET.CodeGeneration as a completely separate library. This way, we can cleanly add a dependency to that package in the build project, keep the rest as-is, and also make code generation from obo files usable in other projects. Additionally, we do not bloat this repo further with another project.
> 
> I think the only thing to adapt code-wise would be parameterizing the namespace of the generated files.
> 
> What do you think @omaus?

Closed due to this and since the functionality is already realized in [OBO.NET](https://github.com/CSBiology/OBO.NET/pull/32).

**Pull request closed without merging**